library/firewall_lib.py

```diff
@@ -317,6 +342,22 @@ def create_service(module, fw, service):
     return fw_service, fw_service_settings
 
 
+def create_ipset(module, fw, ipset, ipset_type):
+    if not ipset_type:
+        module.fail_json(msg="ipset_type needed when creating a new ipset")
+    if not module.check_mode:
+        fw_ipset_settings = FirewallClientIPSetSettings()
+        fw_ipset_settings.setType(ipset_type)
+        fw.config().addIPSet(ipset, fw_ipset_settings)
+        fw_ipset = fw.config().getIPSetByName(ipset)
+        fw_ipset_settings = fw_ipset.getSettings()
+    else:
+        fw_ipset = None
+        fw_ipset_settings = FirewallClientIPSetSettings()
+        fw_ipset_settings.setType(ipset_type)
```

Comment:
Does this actually change anything in check mode?

Reply:
This avoids calling fw.config().addIPSet(), which adds the ipset to permanent configuration.

This organization does create some redundant code though, so I'll refactor to remove that redundancy.